test/stress/pool-create-hanging.js

```diff
@@ -0,0 +1,193 @@
+    debug('========== Reconnecting mssql')
+
+    // this test needs toxiproxy to run so docker mssql is also required
+    mssqlCon = new mssql.ConnectionPool({
+      port: 21433,
+      user: 'sa',
+      password: 'S0meVeryHardPassword',
+      server: 'localhost',
+      requestTimeout: 5000,
+      connectionTimeout: 7000,
+      debug: true,
+      pool: {
+        debug: true,
+        min: 1,
+        max: 1,
+        acquireTimeoutMillis: 10000,
+        testOnBorrow: true
+      }
+    })
+
+    mssqlCon.connectionNumber = connectionNumber += 1
+
+    // NOTE: this never seems to fire
+    mssqlCon.on('error', err => {
+      debug('There was fatal connection error %O', err)
+    })
+
+    debug('Waiting for connection')
+    await mssqlCon.connect()
+  }
+
+  debug('>>>>>>>>>>>>>>>> Running query >>>>>>>>>>>>>>>')
+  debug('%o', {
+    peding: mssqlCon.pool.pending,
+    size: mssqlCon.pool.size,
+    available: mssqlCon.pool.available,
+    spareResourceCapacity: mssqlCon.pool.spareResourceCapacity,
+    borrowed: mssqlCon.pool.borrowed
+  })
+  const res = await mssqlCon.request().query(sql)
+  debug('query response %o', res.recordset)
+  debug('<<<<<<<<<<<<<<<<< query  done <<<<<<<<<<<<<<<<')
+  return res
+}
+
+// setup counters to check queries / results / errors made since last report
+const counters = {}
+function setQueryCounters (name) {
+  counters[name] = {queries: 0, results: 0, errors: 0}
+}
+setQueryCounters('mssql')
+
+// start printing out counters
+let lastCounters = cloneDeep(counters)
+let hangRounds = 0
+let statsInterval = setInterval(() => {
+  const reqsPerSec = {}
+  for (let key of Object.keys(counters)) {
+    reqsPerSec[key] = {
+      queries: (counters[key].queries - lastCounters[key].queries),
+      results: (counters[key].results - lastCounters[key].results),
+      errors: (counters[key].errors - lastCounters[key].errors)
+    }
+  }
+
+  console.log('------------------------ REQUESTS SINCE LAST PRINT ------------------------')
+  if (mssqlCon.pool) {
+    console.dir({
+      peding: mssqlCon.pool.pending,
+      size: mssqlCon.pool.size,
+      available: mssqlCon.pool.available,
+      borrowed: mssqlCon.pool.borrowed
+    })
+  }
+  console.dir(reqsPerSec, { colors: true })
+  console.log('------------------------------------ EOS ---------------------------------')
+  lastCounters = cloneDeep(counters)
+
+  if (reqsPerSec.mssql.queries === 0) {
+    hangRounds++
+  } else {
+    hangRounds = 0
+  }
+
+  if (hangRounds === 3) {
+    console.log('Driver did hang exit with error')
+    process.exit(1)
+  }
+}, 500)
+
+/**
+ * Kill old proxy instance and its connections if found and create new
+ * proxy waiting for connections.
+ */
+async function recreateProxy (serviceName, listenPort, proxyToPort) {
+  try {
+    await rp.delete({
+      url: `${toxicli.host}/proxies/${serviceName}`
+    })
+  } catch (err) {}
+
+  const proxy = await toxicli.createProxy({
```

Comment:
maybe all the proxy code would make more sense in a separate helper file to reduce the clutter?..

Reply:
It is already in helper called `recreateProxy`, I find it better to have that code easily accessible than refactored prematurely. 

Also the kind of proxy setup one needs to create depends on code that is being tested (I already wrote this kind of code for multiple DB drivers), so general helper doesn't make sense yet (unless there will be more tests like this using the same proxycreation).